### PR TITLE
add netlify deploy preview

### PR DESCRIPTION
**Pull request by sckott**
Adds a workflow for Netlify deploy previews for pull requests

### CODE REVIEW

File: .github/workflows/deploy-preview.yml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@v4
+
+      - name: Setup R
+        uses: r-lib/actions/setup-r@v2
+        with:
+          use-public-rspm: true
+
+      - uses: r-lib/actions/setup-r-dependencies@v2
```
Comment (sckott) [inline .github/workflows/deploy-preview.yml:42]: @seankross so this setup requires a DESCRIPTION file in the repo - see error https://github.com/fhdsl/personas/actions/runs/9865033089/job/27241162325?pr=1#step:4:145 - you could add a valid desc file or remove this line in the yaml file

Comment (seankross) [reply .github/workflows/deploy-preview.yml:42]: @sckott I added a description file that I think is valid, still failing though.

Comment (sckott) [reply .github/workflows/deploy-preview.yml:42]: fixed, had to update netlify tokens, see link below

**Review by seankross — APPROVED** [review]
Looks great, thanks so much Scott.

**Pull request merged**